packages/react-dom/src/events/__tests__/ChangeEventPlugin-test.internal.js

```diff
@@ -11,212 +11,219 @@
 
 var React = require('react');
 var ReactDOM = require('react-dom');
-var ReactTestUtils = require('react-dom/test-utils');
-// TODO: can we express this test with only public API?
-var ChangeEventPlugin = require('../ChangeEventPlugin').default;
-var inputValueTracking = require('../../client/inputValueTracking');
 
 function getTrackedValue(elem) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  return tracker.getValue();
+  return elem.value;
 }
 
-function setTrackedValue(elem, value) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  tracker.setValue(value);
-}
-
-function setUntrackedValue(elem, value) {
-  var tracker = inputValueTracking._getTrackerFromNode(elem);
-  var current = tracker.getValue();
+var setUntrackedChecked = Object.getOwnPropertyDescriptor(
+  HTMLInputElement.prototype,
+  'checked',
+).set;
 
-  if (elem.type === 'checkbox' || elem.type === 'radio') {
-    elem.checked = value;
-  } else {
-    elem.value = value;
-  }
-  tracker.setValue(current);
-}
+var setUntrackedValue = Object.getOwnPropertyDescriptor(
+  HTMLInputElement.prototype,
+  'value',
+).set;
 
 describe('ChangeEventPlugin', () => {
-  it('should fire change for checkbox input', () => {
+  var container;
+
+  beforeEach(() => {
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+    container = null;
+  });
+
+  fit('should fire change for checkbox input', () => {
     var called = 0;
 
     function cb(e) {
-      called = 1;
+      called++;
       expect(e.type).toBe('change');
     }
 
-    var input = ReactTestUtils.renderIntoDocument(
+    var node = ReactDOM.render(
       <input type="checkbox" onChange={cb} />,
+      container,
     );
 
-    setUntrackedValue(input, true);
-    ReactTestUtils.SimulateNative.click(input);
-
+    setUntrackedChecked.call(node, false);
+    node.dispatchEvent(new Event('click', {bubbles: true, cancelable: true}));
```

Comment:
We should leave a common about jsdom click misbehaving. It's not obvious and doesn't happen in the browser (it's also possible jsdom has fixed this in new versions so we need to make it clear in case somebody later tries to update it, and the test will fail).

Reply:
Can you please elaborate on what a 'common' is?

Reply:
```javascript
node.dispatchEvent(new Event('click'));
expect(called).toBe(0);
```
Is this what you mean?

Reply:
Sorry, autocorrect. Meant "we should leave a comment". Like in the sample I wrote above.